nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -218,7 +218,7 @@ def main() -> None:
         sdboot_status = subprocess.check_output(["@systemd@/bin/bootctl", "--path=@efiSysMountPoint@", "status"], universal_newlines=True)
 
         # See status_binaries() in systemd bootctl.c for code which generates this
-        m = re.search("^\W+File:.*/EFI/(BOOT|systemd)/.*\.efi \(systemd-boot (\d+)\)$",
+        m = re.search("^\W+File:.*/EFI/(BOOT|systemd)/.*\.efi \(systemd-boot ([0-9.]+)\)$",
```

Comment:
```suggestion
        m = re.search("^\W+File:.*/EFI/(BOOT|systemd)/.*\.efi \(systemd-boot (\d\.\d+)\)$",
```

Reply:
sounds like a bad idea, our system has:
```
File: └─/EFI/systemd/systemd-bootx64.efi (systemd-boot 247.4-2-arch)
```

it's a lustrated system (from arch), maybe matching `systemd-boot (\d+\.\d+)[^)]*\)$` is better

Reply:
@SuperSandro2000 That's not going to work - There can be multiple digits before the `.`, and I don't know that systemd's versioning scheme guarantees exactly one `.`. Seems reasonable to allow for more version levels, especially since they're just being compared with string ordering.

@pennae Good point, I had a similar one installed earlier today. I'll add the `[^)]*` :+1: